src/components/login/Login.js

```diff
@@ -0,0 +1,210 @@
+import {
+  Box,
+  Button,
+  Grid,
+  InputAdornment,
+  TextField,
+  Typography,
+  CircularProgress,
+} from '@material-ui/core';
+import { Link, useHistory } from 'react-router-dom';
+import React, { useState } from 'react';
+import AssignmentIndIcon from '@material-ui/icons/AssignmentInd';
+import IconButton from '@material-ui/core/IconButton';
+import VisibilityIcon from '@material-ui/icons/Visibility';
+import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
+import { create } from '../../helpers/fetchApi';
+import { useSetRecoilState } from 'recoil';
+import informacionUsuarioState from '../../state/login';
+import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+
+export default function Login({ setEstaAutorizado }) {
+  const history = useHistory();
+  const classes = useStyles();
+
+  const [showPassword, setshowPassword] = useState(false);
+  const [iconoCargando, setIconoCargando] = useState(false);
+  const [tengoErrorEn, setTengoErrorEn] = useState({
+    dni: false,
+    contrasenia: false,
+    mandarError: false,
+  });
+
+  const [valoresEntrantes, setValoresEntrantes] = useState({
+    dni: '',
+    contrasenia: '',
+  });
+
+  const setInfoUsuario = useSetRecoilState(informacionUsuarioState);
+
+  const handleChange = (e) => {
+    setValoresEntrantes({
+      ...valoresEntrantes,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleClickShowPassword = () => {
+    setshowPassword(!showPassword);
+  };
+
+  const irARegistro = () => {
+    history.push('/registro');
+  };
+
+  //VALIDAR LOGIN.
+  const validarLogin = async () => {
+    setIconoCargando(true);
+    setTengoErrorEn({ ...tengoErrorEn, mandarError: false });
+
+    await sleep(3000);
+
+    create('/usuarios/login', valoresEntrantes)
+      .then((res) => {
+        setInfoUsuario({
+          token: res.token,
+          nombre: res.nombre,
+          apellido: res.apellido,
+          dni: res.dni,
+        });
```

Comment:
¿Por qué no directamente guardar todo lo que viene de la API?

```suggestion
        setInfoUsuario(res);
```

Reply:
Para no guardar cosas de mas pero ahí está